### PR TITLE
feat(gum): add gum spec

### PR DESCRIPTION
**Pull request by Fall1ngStar**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
New spec for `gum`

**What is the current behavior? (You can also link to an open issue here)**

Missing spec for `gum` (#1547)

**What is the new behavior (if this is a feature change)?**

Add spec for `gum`

**Additional info:**

### CODE REVIEW

File: src/gum.ts [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+];
+
+const styleOptionGenerator = (element: string) =>
+  styleOptions.map(
+    (style): Fig.Option => ({
+      ...style,
+      name: (style.name as string).replace("--", `--${element}.`),
+      description: `${style.description} for ${element}`,
+    })
+  );
```
Comment (SeparateRecords) [inline src/gum.ts:46]: This is really smart. Nice work 🤩

File: src/gum.ts [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+      args: {
+        name: "options",
+        isVariadic: true,
+        description: "Options to choose from",
+      },
+      options: [
+        {
+          name: "--height",
+          description: "Height of the list",
+          args: {},
```
Comment (SeparateRecords) [inline src/gum.ts:64]: Could you add a `name` property to the arg object? Same comment for all the args below. If no obvious name for the value jumps out, don't worry about it, but for example with this you could do `{ name: "lines" }`

File: src/gum.ts [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+    {
+      name: "confirm",
+      description: "Ask a user to confirm an action",
+      args: {
+        name: "prompt",
+        description: "Prompt to display",
+      },
+      options: [
+        {
+          name: "--afirmative",
```
Comment (SeparateRecords) [inline src/gum.ts:116]: Quick typo fix
```suggestion
          name: "--affirmative",
```

File: src/gum.ts [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+  ],
+  options: [
+    {
+      name: ["--help", "-h"],
+      description: "Show context-sensitive help",
+    },
+    {
+      name: ["--version", "-v"],
+      description: "Print the version number",
+    },
```
Comment (SeparateRecords) [inline src/gum.ts:382]: These options are available on all child commands, you can actually add `isPersistent: true` to make them inherited!

File: src/gum.ts [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+  ],
+  // Only uncomment if gum takes an argument
+  // args: {}
```
Comment (SeparateRecords) [inline src/gum.ts:385]: Can be deleted

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
Awesome. Thank you so much for working on this! Just a couple of (hopefully) easy changes then it's ready to merge 🙂

File: src/gum.ts (hunk 1 above)
Comment (Fall1ngStar) [reply src/gum.ts:46]: 😊 thank !

File: src/gum.ts (hunk 2 above)
Comment (Fall1ngStar) [reply src/gum.ts]: Done, hope the naming I gave is matching the Fig style

File: src/gum.ts (hunk 3 above)
Comment (Fall1ngStar) [reply src/gum.ts]: Fixed

File: src/gum.ts (hunk 4 above)
Comment (Fall1ngStar) [reply src/gum.ts]: Fixed

**Review by SeparateRecords — APPROVED** [review]
Amazing, thank you so much @Fall1ngStar!

**Pull request merged**